User ebuild's ${T} instead of /tmp to hide leftover files from ./configure.

'run-ag.sh' does not clean temporary directories after itself and makes
it hard to implement it cleanly without changing 'run-ag.sh' structure.

Reported-by: Fedja Beader
Bug: https://bugs.gentoo.org/704740
--- a/build-aux/run-ag.sh
+++ b/build-aux/run-ag.sh
@@ -45,7 +45,7 @@ find_exe() {
   return 0
 }
 
-STAMP_TEMP_DIR=$(mktemp --suffix=.tdir -d /tmp/run-ag-XXXXXXXX)
+STAMP_TEMP_DIR=$(mktemp --suffix=.tdir -d -t run-ag-XXXXXXXX)
 exec 9>&2 2>> ${STAMP_TEMP_DIR}/mk-stamps.log
 VERBOSE=1
 
--- a/config/mk-shdefs.in
+++ b/config/mk-shdefs.in
@@ -130,7 +130,7 @@
 
 set -x
 exec 9>&2
-tmp=$(mktemp --suffix=.tdir -d /tmp/shdefs-XXXXXXXXX)
+tmp=$(mktemp --suffix=.tdir -d -t shdefs-XXXXXXXXX)
 exec 2>> $tmp/mk-shdef.log
 init $0 $1
 mk_config
